kits/arm/hebi_util.cpp

```diff
@@ -0,0 +1,81 @@
+// hebi_util.cpp
+
+#include "group_command.hpp"
+#include "group_feedback.hpp"
+#include "robot_model.hpp"
+#include "arm/arm.hpp"
+#include "util/file.hpp"
+
+#include <stdexcept>
+#include <filesystem>
+#include <iostream>
+
+#include "hebi_util.hpp"
+
+std::unique_ptr<hebi::util::MobileIO> createMobileIOFromConfig(const hebi::RobotConfig& config, const std::string& config_file_path) {
+    std::map<std::string, std::string> mobile_io_dict;
+    std::vector<std::string> errors;
+
+    hebi::util::file::File cfg_file(config_file_path);
+    auto parent_dir_absolute = cfg_file.getParentDirectory().getAbsolutePath();
+
+    // Lambda function to check file paths
+    auto check_file = [&errors, &parent_dir_absolute](const std::string& type, const std::string& relative_filename) {
+        // Ensure file is relative:
+        if (hebi::util::file::File(relative_filename).isAbsolute()) {
+            errors.push_back("'" + type + "' exists but provides an absolute path.");
+            return std::string();
+        }
+        // Ensure file exists
+        hebi::util::file::File absolute_filename(parent_dir_absolute);
+        absolute_filename.append(relative_filename);
+        if (!absolute_filename.exists()) {
+            errors.push_back("'" + type + "' exists but does not point to a valid file.");
+            return std::string();
+        }
+        // Return success!
+        return absolute_filename.getAbsolutePath();
+    };
+
+    // Validate the mobile_io configuration
+    if (config.getUserData().strings_.count("mobile_io_family") &&
+        config.getUserData().strings_.count("mobile_io_name") &&
+        config.getUserData().strings_.count("mobile_io_layout")) {
+
+        // Check that all required fields are present and are strings
+        if (config.getUserData().strings_.at("mobile_io_family").empty() ||
+            config.getUserData().strings_.at("mobile_io_name").empty()) {
+            errors.push_back("HEBI config \"user_data\"'s \"mobile_io_...\" fields must be non-empty strings.");
```

Comment:
This seems like a fatal error here; should probably return instead of continuing?

Reply:
Ahh -- nevermind; this is fine.  It doesn't hit anything critical before the return below! 👍